dali/ft/daftformat.cpp

```diff
@@ -691,6 +691,31 @@ CCsvPartitioner::CCsvPartitioner(const FileFormat & _format) : CInputBasePartiti
     fields.setown(new KeptAtomTable);
 }
 
+void CCsvPartitioner::getRecordStructure(StringBuffer & _recordStructure)
+{
+    if (nullptr == strstr(recordStructure.str(),"END;"))
+    {
+        const byte *buffer = bufferBase();
+        ensureBuffered(headerSize);
+        assertex((headerSize == 0) || (numInBuffer != bufferOffset));
+
+        bool processFullBuffer = false;
+
+        unsigned size = getSplitRecordSize(buffer+bufferOffset, numInBuffer-bufferOffset, processFullBuffer);
+
+        if (size == 0)
+            throwError1(DFTERR_PartitioningZeroSizedRowLink,((offset_t)(buffer+bufferOffset)));
+
+        if (size > bufferSize)
+        {
+            LOG(MCdebugProgressDetail, unknownJob, "Split record size %d (0x%08x) is larger than the buffer size: %d", size, size, bufferSize);
+            throwError2(DFTERR_WrongSplitRecordSize, size, size);
+        }
```

Comment:
The above code looks same as findSplitPoint()
It's not very clear what it's doing in this context, it would clearer if in method with an appropriate name and called from both.
Or, if not all it applies in this context it should be simplified here, e.g.
i) remove processFullBuffer declaration, 
ii) bufferOffset must always be 0 afaict, so could be simplified/clearer.

I'd probably common and put in readNextSize() or similar